userspace/libpman/src/configuration.c

```diff
@@ -165,15 +169,126 @@ int pman_get_required_buffers()
 	return g_state.n_required_buffers;
 }
 
+static char byte_array[] = "BPF_TRACE_RAW_TP";
+
+bool check_BPF_TRACE_RAW_TP(void)
+{
+	// These locations are taken from libbpf library:
+	// https://elixir.bootlin.com/linux/latest/source/tools/lib/bpf/btf.c#L4767
+	const char *locations[] = {
+		/* try canonical vmlinux BTF through sysfs first */
+		"/sys/kernel/btf/vmlinux",
+		/* fall back to trying to find vmlinux on disk otherwise */
+		"/boot/vmlinux-%1$s",
+		"/lib/modules/%1$s/vmlinux-%1$s",
+		"/lib/modules/%1$s/build/vmlinux",
+		"/usr/lib/modules/%1$s/kernel/vmlinux",
+		"/usr/lib/debug/boot/vmlinux-%1$s",
+		"/usr/lib/debug/boot/vmlinux-%1$s.debug",
+		"/usr/lib/debug/lib/modules/%1$s/vmlinux",
+	};
+	char path[PATH_MAX + 1];
+	struct utsname buf;
+	char *data = NULL;
+	FILE *f = NULL;
+	uname(&buf);
+
+	for (int i = 0; i < sizeof(locations) / sizeof(*locations); i++)
+	{
+		snprintf(path, PATH_MAX, locations[i], buf.release);
+
+		// On success `faccessat` returns 0.
+		if(faccessat(0, path, R_OK, AT_EACCESS) != 0)
+		{
+			// The file doesn't exist, loop on the next one.
+			continue;
+		}
+
+		f = fopen(path, "r");
+		if(!f)
+		{
+			goto loop_again;
+		}
+
+		// Seek to the end of file
+		if(fseek(f, 0, SEEK_END))
+		{
+			goto loop_again;
+		}
+		
+		// Return the dimension of the file
+		long sz = ftell(f);
+		if (sz < 0)
+		{
+			goto loop_again;
+		}
+
+		// Seek again to the beginning of the file
+		if(fseek(f, 0, SEEK_SET))
+		{
+			goto loop_again;
+		}
+
+		// pre-alloc memory to read all of BTF data 
+		data = malloc(sz);
+		if (!data)
+		{
+			goto loop_again;
+		}
+
+		// read all of BTF data
+		if(fread(data, 1, sz, f) < sz)
+		{
+			goto loop_again;
+		}
+
+		// Search 'BPF_TRACE_RAW_TP' byte array
+		int z = 0;
+		for(int j = 0; j< sz; j++)
+		{
+			if(data[j] == byte_array[z])
+			{
+				z++;
+				if(z == sizeof(byte_array) / sizeof(*byte_array))
+				{
+					fclose(f);
+					free(data);
+					return true;
```

Comment:
We can use an auxiliar `bool` variable like `res`, set it to false on instantiation, check it on the loop condition and use it to make this a bit more consistent with the rest of the code.
```suggestion
					res = true;
					goto loop_again;
```
We could also rename `loop_again` to something that is a bit more consistent, if you split this to its own function it could just be something like `end`

Reply:
done :)